Tools-Override/tests.targets

```diff
@@ -55,11 +61,71 @@
 
     <TestCommandLine Condition="'$(Performance)'!='true'">$(TestProgram) $(TestArguments) {XunitTraitOptions}</TestCommandLine>
   </PropertyGroup>
+  
+  <!-- This will generate a version of the dotnet cli that runs with the live
+       build of corefx using the other components gathered from the stable 
+       version consumed from the Tools directory
+       To debug any issues set COREHOST_TRACE=1 and rerun dotnet
+  -->
+  <UsingTask TaskName="GenerateDepsJson" AssemblyFile="$(BuildToolsTaskDir)CoreFx.Tools.dll"/>
+  <PropertyGroup>
+    <NewSharedFxVersion>9.9.9</NewSharedFxVersion>
+    <TestDotNetCliPath>$(ToolsDir)dotnetcli\shared\Microsoft.NETCore.App\$(NewSharedFxVersion)</TestDotNetCliPath>
+    <OldDotNetCliPath>$(ToolsDir)dotnetcli</OldDotNetCliPath>
+  </PropertyGroup>
+  <Target Name="ConstructTestingHost" Condition="!Exists('$(TestDotNetCliPath)')" >
+    <ItemGroup>
+      <LiveDotNetCliComponents Include="$(RuntimePath)\*.*" Exclude="$(RuntimePath)\xunit.console.netcore.exe" />
+      <DotNetCliSharedFrameworkSdkRuntimeConfig Include="$(OldDotNetCliPath)\sdk\**\dotnet.runtimeconfig.json" />
+      <DotNetCliHostPolicy Include="$(OldDotNetCliPath)\shared\**\hostpolicy.dll" />
+      <XunitRuntimeConfig Include="$(ObjDir)xunit\xunit.console.netcore.runtimeconfig.json" />
+    </ItemGroup>
+    <ReadLinesFromFile File="@(DotNetCliSharedFrameworkSdkRuntimeConfig)" >  
+        <Output TaskParameter="Lines"  
+                ItemName="LinesFromDotNetCliSharedFrameworkSdkRuntimeConfig"/> 
+    </ReadLinesFromFile>
+    <ItemGroup>
+      <LinesToWriteDotNetCliSharedFrameworkSdkRuntimeConfig Include="$([System.String]::Copy('%(LinesFromDotNetCliSharedFrameworkSdkRuntimeConfig.Identity)').Replace('1.0.0','9.9.9'))" />
+      <LinesToWriteDotNetCliSharedFrameworkSdkRuntimeConfig Include="};}" />
+    </ItemGroup>
+    <MakeDir  Directories="$(ObjDir)xunit\"/>  
+    <WriteLinesToFile  File="$(ObjDir)xunit\xunit.console.netcore.runtimeconfig.json"  
+                       Lines="@(LinesToWriteDotNetCliSharedFrameworkSdkRuntimeConfig)"  
+                       Overwrite="true" />
+    <!-- Currently need a hostpolicy within the shared framework, reusing the one in the Tools dir -->
+    <Copy SourceFiles="@(DotNetCliHostPolicy)" 
+          DestinationFiles="@(DotNetCliHostPolicy -> '$(TestDotNetCliPath)\%(Filename)%(Extension)')" />
+    <Copy SourceFiles="@(LiveDotNetCliComponents)" 
```

Comment:
No I don't want to gate on that as we want the contents to be constructed by our source build and expect they will work. If they don't it will fail during some test run.